### DISCUSSION TITLE
Future of this Repo and my setup - Feel free to share your thoughts

### DISCUSSION THREAD

**Discussion by SimpleHomelab** [htpcBeginner/docker-traefik#125] [Ideas]
Dear All, 

Thank you for actively participating in the community. I apologize for being quite silent or not being very active.

I have been working on heavily on the next evolution of my setup. Here are somethings I plan to work on in the coming months: 

- [x] Switch my server to Proxmox VE - this would allow me to spin up new VMs and containers for testing without affecting my main setup - this is already well in progress. I had a steep learning curve for Proxmox that needed a lot of attention.
- [x] Setup a cloud/seed box with Rclone + MergerFS for my media hosted in Google Drive - This is complete.
- [ ] Move most of my apps over from my home server to the cloud/seed box. 
- [ ] Move my home server to Proxmox and setup remaining apps need for Home (home assistant, motioneye, etc.). 
- [ ] Look into moving my website over to the cloud/seed box to - for central management and also reduce costs.
- [ ] 6. Explore and learn Kubernetes and move my setup over to it. 

I am always actively working on my setup even though you may not see commits here. 

Please feel free to share any thoughts or comments.

**Comment by SimpleHomelab** [comment]
Pulling in a few folks who have been active here: @shawnthompson @gzecchi @robflate

**Reply by gzecchi** [reply]
Hi @htpcBeginner,

glad to help, just let me know in which part 👍 

In my home server, I'm also running NextCloud and BitWarden.

Regarding Kubernetes (k8s), be aware that requires some effort to configure and maintain a self-hosted installation based on a Proxmox VM. Configuring Traefik as K8s Ingress is a bit tricky but I've already done it so I can help.

Please keep also in mind the Storage Persistency of the DB Data across the k8s workers or the DB replica.

GZ

**Reply by creativerbl** [reply]
@gzecchi - Would you mind sharing your docker-compose for Bitwarden?

**Comment by shawnthompson** [comment]
Thanks for pulling me in, I don't think I'll be much help but I'm still trying to get my setup going, although I haven't spend much time on it lately. A lot of this stuff is over my head but I'm learning as I got. I'll try to contribute whenever I can.

Thanks again for this great setup!

**Comment by uasdj25** [comment]
In my experience, Kubernetes is a bit of a beast.  It'd be quite a migration, but interesting.  I think it's certainly more widely adopted in my industry, but I am not sure I'm anxious yet to make that switch.  The complicating factor, of course, is that K8 is deprecating the Docker engine, and I'm hesitant to move away from Docker completely as I'm a big fan.  I am very, very interested in seeing how your transition from home server to cloud based server (with the exception of home apps) goes.  I've considered this before, but I wasn't sure about how complex or cost effective it would really be.  I look forward to learning more on that topic for sure.

**Comment by Maynardnaze** [comment]
Thanks for all the hard work!

**Comment by PrplHaz4** [comment]
Some thoughts based on how my (initially similar) setup has been evolving...time to manage and maintain is reduced...etc.  Thanks for all the great great insight over the years!

1. **Proxmox** - I'm trying to remove this or relegate it's role to test setups only - lately it feels like just another unnecessary maintenance item and shutting down all the services on it to install updates is disruptive - I'll be replacing the Debian VM with BurmillaOS and dedicate that machine to running docker containers (which is basically what it does today with 2 extra layers - Proxmox and Debian.
2. **Media** - I'm moving away from most locally-hosted media/downloading and replacing with paid services and cheap streaming boxes/sticks/apps
3. **Docs/Photos** - I treat these as requiring the highest level of resiliency (on btrfs raid in OMV + snapshots + backups) and sync to Google Drive for better usability/family access
4. **Centralized Logging** - seems to be missing from your setup - I'm running ElasticStack which requires a lot more maintenance than I'd prefer but supports more built-in processors and dashboards than the others (Influxdb v2, Loki, Graylog) which keeps me from having to produce and maintain my own.
5. **Wildcard Certs** - I appreciate the obscurity of non-enumerated domains in CF DNS
6. **Portainer Agent** - I use this to manage multiple docker hosts in a single portainer instance
7. **Future Traefik** - I'd like to maintain a single Traefik ingress for all docker hosts on my local network - likely requires Swarm or k8s
8. **Kubernetes** - Every time I scratch this itch, it seems like a ton of added complexity for little benefit (where most services I run do not benefit from redundancy and are not written to take advantage, or require node-specific hardware )
9. **Harvester** - Possible migration path away from Proxmox/Docker towards better gitops, k8s, converged storage...etc https://github.com/harvester/harvester
